applications/DemStructuresCouplingApplication/python_scripts/dem_fem_coupling_algorithm.py

```diff
@@ -0,0 +1,259 @@
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Dem.ELASTIC_FORCES)
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Dem.CONTACT_FORCES)
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Dem.TANGENTIAL_ELASTIC_FORCES)
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Dem.SHEAR_STRESS)
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Dem.NON_DIMENSIONAL_VOLUME_WEAR)
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Dem.IMPACT_WEAR)
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Dem.PARTICLE_ROTATION_ANGLE)
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Kratos.DELTA_ROTATION)
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Kratos.ANGULAR_VELOCITY)
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Dem.LOCAL_ANGULAR_VELOCITY)
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Dem.LOCAL_AUX_ANGULAR_VELOCITY)
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Dem.ORIENTATION_REAL)
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Dem.ORIENTATION_IMAG)
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Dem.ORIENTATION)
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Dem.AUX_ORIENTATION)
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Kratos.ANGULAR_MOMENTUM)
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Dem.RIGID_ELEMENT_FORCE)
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Dem.PARTICLE_MOMENT)
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Kratos.EXTERNAL_APPLIED_FORCE)
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Kratos.EXTERNAL_APPLIED_MOMENT)
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Dem.PRINCIPAL_MOMENTS_OF_INERTIA)
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Dem.CLUSTER_VOLUME)
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Kratos.NODAL_MASS)
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Dem.CHARACTERISTIC_LENGTH)
+        self.structural_solution._GetSolver().main_model_part.AddNodalSolutionStepVariable(Dem.PARTICLE_DENSITY)
```

Comment:
Do we really need to add all these variables to the structure??

Reply:
Without that many variables included, the program shows memory-releasing problems after finishing the simulations. I could remove those variables and have a closer look at what is really happening.

Reply:
In fact, they are the same that are added in the AddRigidFaceVariables function in DEM Procedures.

Reply:
I think this is totally wrong and we have to know what are the variables that are actually needed.
It makes no sense to add those variables without a reason. DEM_procedures is adding those variables to the DEM model parts because those variables are needed on the nodes. The structure does not need all those variables. Maybe one? Let's fix this as soon as possible, it's ridiculous to add so many useless variables for no reason.

Reply:
The RigidFaces of the DEM don't need all those variables either.

Reply:
Maybe some of them are not needed (e.g. CLUSTER_VOLUME), but others are needed if the rigid element rotates as a free body, isn' it?

Reply:
Thanks, @joaquinirazabal 
I think it is not exactly as you say. I guess that the rigid element features will not be necessary in this case, as the RigidFace elements just follow the skin of the Structure as it deforms. They are not free as they would be in a solid rigid context.
@maceligueta I'll fix that as soon as possible

Reply:
@joaquinirazabal . I understand that all those variables should be in the central node of the rigid body, but I think that the nodes of the walls do not need the variables. Are all those nodes mixed with the central nodes in the same model part?

Reply:
Let's talk about this during the meeting on Friday, if you are available. For the moment, I would avoid adding the variables to the structure.

Reply:
Ok, yes I would be available. I agree that this should not be added to the structure and it would be interesting to think about improving it for rigid bodies.

Reply:
Thanks, @maceligueta @joaquinirazabal
Anyway, just to clarify... Am I right when I say that no RBE related code is necessary in these simulations? We are only following skins, right?

Reply:
Yes, I think so @salvalatorre, I was only talking about RBEs. Sorry if I messed up the conversation.